loris/authorizer.py

```diff
@@ -276,16 +283,17 @@ def _roles_from_request(self, request):
                 logger.debug(value)
                 raise AuthorizerException(message="invalidCredentials: expired")
         else:
-            cval = cval.encode('utf-8')
-            key = base64.urlsafe_b64encode(self.kdf().derive(secret.encode('utf-8')))
+            key = base64.urlsafe_b64encode(self.kdf().derive(secret))
             fern = Fernet(key)
             value = fern.decrypt(cval)
-            if not value.startswith(origin):
+            if not value.startswith(origin.encode('utf8')):
                 # Cookie/Token theft
                 return []
             else:
-                value = value[len(origin)+1:]
+                value = value[len(origin.encode('utf8'))+1:]
 
+        if isinstance(value, bytes):
```

Comment:
EAFP suggests it might be preferable to write:

```python
try:
    value.decode('utf8')
except AttributeError:
    pass
```

but I don't think it makes a huge amount of difference.